### PR TITLE
SDK-1007: Added enum consts for the entities

### PR DESCRIPTION
**Pull request by vol4onok**
https://spryker.atlassian.net/browse/SDK-1007

### CODE REVIEW

**Review by sergeyspryker — DISMISSED** [review]

File: src/Enum/Setting.php [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+     * @var string
+     */
+    public const PATH_REPORT_USAGE_STATISTICS = 'report_usage_statistics';
+
+    /**
+     * Path setting key for core namespaces.
+     *
+     * @var string
+     */
+    public const PATH_CORE_NAMESPACE = 'coreNamespaces';
```
Comment (dereuromark) [inline src/Enum/Setting.php]: Inconsistent usage of Singular Plural

File: src/Enum/Setting.php [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+     * @var string
+     */
+    public const PATH_PROJECT_NAMESPACES = 'projectNamespaces';
+
+    /**
+     * Path setting key for default violation format.
+     *
+     * @var string
+     */
+    public const PATH_DEFAULT_VIOLATION_OUTPUT_PATH = 'default_violation_output_format';
```
Comment (dereuromark) [inline src/Enum/Setting.php]: Double path is maybe not a good naming

File: src/Enum/Task.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+     * @var string
+     */
+    public const TASK_TYPE_LOCAL_CLI = 'local_cli';
+
+    /**
+     * Local cli task type.
+     *
+     * @var string
+     */
+    public const TASK_SET_TYPE = 'task_set';
```
Comment (dereuromark) [inline src/Enum/Task.php]: Isnt this wrong order? Or looks a bit inconsistent

File: src/Enum/ValueTypeEnum.php [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+ */
+
+namespace SprykerSdk\SdkContracts\Enum;
+
+final class ValueTypeEnum
+{
+    /**
+     * @var string
+     */
+    public const TYPE_BOOLEAN = 'boolean';
```
Comment (dereuromark) [inline src/Enum/ValueTypeEnum.php]: Inconsistent
Why int, but not bool?

Comment (vol4onok) [reply src/Enum/ValueTypeEnum.php]: Seems like it comes from gettype(). changed to `integer`

File: src/Enum/Task.php [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+    /**
+     * Local cli task type.
+     *
+     * @var string
+     */
+    public const TYPE_LOCAL_CLI = 'local_cli';
+
+    /**
+     * Local cli task type.
```
Comment (pavelmaksimov25) [inline src/Enum/Task.php]: the description doesn't match the constant and value.

File: src/Enum/ValueTypeEnum.php [hunk 6]
```diff
@@ -32,5 +32,5 @@ final class ValueTypeEnum
     /**
      * @var string
      */
-    public const TYPE_INT = 'int';
+    public const TYPE_INTEGER = 'integer';
```
Comment (dereuromark) [inline src/Enum/ValueTypeEnum.php]: Please rather use the short form for bool above! We never use the long versions on code elements as the short ones are clear enough already. They are only for textual usage.

**Review by pavelmaksimov25 — APPROVED** [review]

File: src/Enum/ValueTypeEnum.php (hunk 6 above)
Comment (vol4onok) [reply src/Enum/ValueTypeEnum.php]: changed

**Review by sergeyspryker — APPROVED** [review]

File: src/Enum/Setting.php [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+     * @var string
+     */
+    public const PATH_PROJECT_NAMESPACES = 'projectNamespaces';
+
+    /**
+     * Path setting key for default violation format.
+     *
+     * @var string
+     */
+    public const PATH_DEFAULT_VIOLATION_OUTPUT_FORMAT = 'default_violation_output_format';
```
Comment (dereuromark) [inline src/Enum/Setting.php:80]: I am a bit confused about the different casings
camelBacked 'projectNamespaces' vs underscored 'default_violation_output_format'

I feel like this should be harmonized if you make this now an explicit cross-repo dependency for long term constant usage.

**Review by dereuromark — APPROVED** [review]
Apart from my comment

**Comment by pavelmaksimov25** [comment]
Please,  do a dev qa, add `qa-done` label and ping me when it's done

**Comment by dereuromark** [comment]
You ignored my comment, though.

**Pull request merged**